ote/src/clj/ote/transit_changes/detection.clj

```diff
@@ -196,7 +196,7 @@
   (->> (if beginning?
          weekhash
          (reverse weekhash))
-       (take-while nil?)
+       (take-while #(or (nil? %) (keyword? %)))
```

Comment:
This might be too simple solution, but lets have a look how well it works. No need to make this complicated if simpler is good enough.

Reply:
What this does, is it makes the logic to treat holidays like no-traffic days.
The current problems are solved, because logic does not think anymore that holidays before route start or after route end, are traffic.
Currently logic already handles time before/after traffic as no-traffic, so holidays will blend in there.

The downside in this might be that a no-traffic+holidays will in some cases reach the no-traffic reporting threshold, when the no-traffic alone would not have done that.

The positive side is, a no-traffic with holiday in between is handled as one big no-traffic period, instead of two short ones.

Reply:
What this does, is it makes the logic to treat holidays like no-traffic days.
The current problems are solved, because logic does not think anymore that holidays before route start or after route end, are traffic.
Currently logic already handles time before/after traffic as no-traffic, so holidays will blend in there.

The downside in this might be that a no-traffic+holidays will in some cases reach the no-traffic reporting threshold, when the no-traffic alone would not have done that.

The positive side is, a no-traffic with holiday in between is handled as one big no-traffic period, instead of two short ones.